plugins/catalog-backend-module-msgraph/package.json

```diff
@@ -33,7 +33,7 @@
     "start": "backstage-cli package start"
   },
   "dependencies": {
-    "@azure/msal-node": "^1.1.0",
+    "@azure/identity": "^2.1.0",
```

Comment:
@Rugvip A couple of other plugins reference patch versions of 2.0.x - do those need to be upgraded to 2.1 to be compatible with these changes?
* `kubernetes-backend` : `^2.0.4`
* `techdocs-node`: `^2.0.1`

I could get away keepign to version 2.0.4 in this plugin, however 2.1 provides a far superior error message when credentials aren't available.

In 2.0.4, it just reports the last error in the credential change

> type=plugin class=MicrosoftGraphOrgEntityProvider taskId=MicrosoftGraphOrgEntityProvider:default:refresh taskInstanceId=3a0dc8a7-342d-49e0-846b-46ddbc6487ce stack=Error: ERROR: No subscription found. Run 'az account set' to select a subscription.

However 2.1.0 reports all errors in the credential chain, making all the available options clearer.

> type=plugin class=MicrosoftGraphOrgEntityProvider taskId=MicrosoftGraphOrgEntityProvider:default:refresh taskInstanceId=c9067854-001c-4dc7-91e8-c1a5eecf60c6 errors=
CredentialUnavailableError: EnvironmentCredential is unavailable. No underlying credential could be used. To troubleshoot, visit https://aka.ms/azsdk/js/identity/environmentcredential/troubleshoot.,
CredentialUnavailableError: ManagedIdentityCredential: Authentication failed. Message ManagedIdentityCredential - No MSI credential available,CredentialUnavailableError: No implementation of `VisualStudioCodeCredential` is available. You must install the identity-vscode plugin package (`npm install --save-dev @azure/identity-vscode`) and enable it by importing `useIdentityPlugin` from `@azure/identity` and calling `useIdentityPlugin(vsCodePlugin)` before creating a `VisualStudioCodeCredential`.,CredentialUnavailableError: ERROR: No subscription found. Run 'az account set' to select a subscription.
CredentialUnavailableError: No implementation of `VisualStudioCodeCredential` is available. You must install the identity-vscode plugin package (`npm install --save-dev @azure/identity-vscode`) and enable it by importing `useIdentityPlugin` from `@azure/identity` and calling `useIdentityPlugin(vsCodePlugin)` before creating a `VisualStudioCodeCredential`.
CredentialUnavailableError: ERROR: No subscription found. Run 'az account set' to select a subscription.
CredentialUnavailableError: The 'Az.Account' module >= 2.2.0 is not installed. Install the Azure Az PowerShell module with: "Install-Module -Name Az -Scope CurrentUser -Repository PSGallery -Force".

Reply:
The `^` range includes minor versions while above `0.x`, so `^2.0.4` matches `^2.1.0`, although `^0.2.4` does not match `0.3.0`. It doesn't hurt to bump them though just to make sure people are getting those improvements.

Reply:
@afscrome @Rugvip we've recently updated backstage to 1.4.0 and we've been expericing intermittent issues with Azure token ever since. After some troubleshooting I found the root cause to be version 2.1.0 of this package (more on https://github.com/Azure/azure-sdk-for-js/issues/22722)

For now I just pinned our instance to use `"@azure/identity": "2.0.5"`, but I'm afraid other people will face this issue as they upgrade to latest. I can PR a change to downgrade it if you both think this is the right move.